packages/components/rating/src/rating.tsx

```diff
@@ -0,0 +1,105 @@
+import {forwardRef} from "@nextui-org/system";
+import {useMemo} from "react";
+import {VisuallyHidden} from "@react-aria/visually-hidden";
+import {RadioGroup} from "@nextui-org/radio";
+
+import {UseRatingProps, useRating} from "./use-rating";
+import RatingSegment from "./rating-segment";
+import {RatingProvider} from "./rating-context";
+
+export interface RatingProps extends UseRatingProps {}
+
+const Rating = forwardRef<"div", RatingProps>((props, ref) => {
+  const context = useRating({...props, ref});
+
+  const {
+    Component,
+    children,
+    length,
+    hasHelper,
+    isInvalid,
+    description,
+    errorMessage,
+    setRatingValue,
+    getBaseProps,
+    getMainWrapperProps,
+    getIconWrapperProps,
+    getInputProps,
+    getHelperWrapperProps,
+    getDescriptionProps,
+    getErrorMessageProps,
+    value,
+    ratingValue,
+    name,
+    onBlur,
+    onChange,
+  } = context;
+
+  const IconList = useMemo(() => {
+    if (children) {
+      return <div {...getIconWrapperProps()}>{children}</div>;
+    }
+
+    return (
+      <div {...getIconWrapperProps()}>
+        <RadioGroup name={name} orientation="horizontal" onBlur={onBlur} onChange={onChange}>
+          {Array.from(Array(length)).map((_, idx) => (
+            <RatingSegment key={"segment-" + idx} index={idx} />
+          ))}
+        </RadioGroup>
+      </div>
+    );
+  }, [children, length, getIconWrapperProps, name, onBlur, onChange]);
+
+  const Helper = useMemo(() => {
+    if (!hasHelper) {
+      return null;
+    }
+    if (isInvalid && !!errorMessage) {
+      return (
+        <div {...getHelperWrapperProps()}>
+          <div {...getErrorMessageProps()}>{errorMessage}</div>
+        </div>
+      );
+    }
+
+    return (
+      <div {...getHelperWrapperProps()}>
+        <div {...getDescriptionProps()}>{description}</div>
+      </div>
+    );
+  }, [
+    hasHelper,
+    isInvalid,
+    description,
+    errorMessage,
+    getHelperWrapperProps,
+    getDescriptionProps,
+    getErrorMessageProps,
+  ]);
+
+  const Input = useMemo(
+    () => (
+      <VisuallyHidden>
+        <input {...getInputProps()} />
+      </VisuallyHidden>
+    ),
+    [value, ratingValue, setRatingValue, getInputProps],
+  );
```

Comment:
Why is this `Input` necessary?

Reply:
This is for supporting the react-hook-form, this is where the `register` props will be passed on.

Reply:
`Proper usage of RadioGroup should naturally address this issue.`
Seems like I was not using the radio group in correct way. Remove the input in 4fca944
Thanks for the suggestion.

Reply:
Seems like we do need input. 
On checking again, without input the defaultValues of the hook-form does not function properly. Since, the radioGroup component does not store the selectedValue, we would need to have input component on the rating side which will store the selectedValue.
This input component will get the ref from the `register` of hook-form and its value will be the selectedValue.